app/controllers/orders_controller.rb

```diff
@@ -27,4 +28,19 @@ def create
     end
   end
 
+  def cancel
+    Order.update(params[:order_id], status: "cancelled")
+    redirect_back(fallback_location: root_path)
+  end
+
+  def paid
+    Order.update(params[:order_id], status: "paid")
+    redirect_back(fallback_location: root_path)
+  end
+
+  def completed
+    Order.update(params[:order_id], status: "completed")
+    redirect_back(fallback_location: root_path)
+  end
```

Comment:
I'm so on the fence about this. It's not ReSTful, but it's very clear what you're doing here, and I really like that you understand the routes file/controller#actions/controller well enough that you can implement this.

That said, in a larger application this could cause your controller to balloon as someone else in your team sees this pattern and continues to add to it as other ways to change an order are introduced. In particular, this is interesting because you're using `redirect_back`, which I don't think is what you'd want to do if you added an `order#update` controller action.

One alternative that you might consider: pass the new status as a query param. That would at least allow you to clean up some of the duplication here:

```
def update_status
  Order.update(params[:order_id], status: params[:status])
  redirect_back(fallback_location: root_path)
end
```

Alternatively, you could use the `update` controller action with the same query params, but you would need to figure out how to handle the different redirects you want to achieve.